gammapy/datasets/map.py

```diff
@@ -408,12 +409,17 @@ def npred_background(self):
             Predicted counts from the background.
         """
         background = self.background
-
         if self.background_model and background:
             values = self.background_model.evaluate_geom(geom=self.background.geom)
-            background = background * values
+            if self._background_cached is None:
+                self._background_cached = background * values
```

Comment:
This caching does not make sense to me. Don't you want to check on the model parameters and avoid the recomputation if not needed?

Reply:
yes that could be added for now I just did that as most of the time is spend in copy during `background = background * values` just avoiding this allow to reduce npred_background computation by about 40%.

Reply:
added